pqtgo/generator.go

```diff
@@ -310,6 +310,14 @@ type %sIterator struct {
 	rows *sql.Rows
 	cols []string
 }
+type I%sIterator interface {
+	Next() bool
+	Close() error
+	Err() error
+	Columns() ([]string, error)
+	%s() (*%sEntity, error)
+}
+var _ I%sIterator = (*%sIterator)(nil)
```

Comment:
check on compile time that struct fits the interface, I have found this approach in Google libraries, like this https://github.com/grpc/grpc-go/blob/master/server.go#L549